clients/src/main/java/org/apache/kafka/clients/admin/internals/RemoveMembersFromConsumerGroupHandler.java

```diff
@@ -79,55 +90,82 @@ public String apiName() {
         Set<CoordinatorKey> groupIds,
         AbstractResponse abstractResponse
     ) {
+        validateKeys(groupIds);
+
         final LeaveGroupResponse response = (LeaveGroupResponse) abstractResponse;
-        Map<CoordinatorKey, Map<MemberIdentity, Errors>> completed = new HashMap<>();
-        Map<CoordinatorKey, Throwable> failed = new HashMap<>();
-        List<CoordinatorKey> unmapped = new ArrayList<>();
+        final Map<CoordinatorKey, Map<MemberIdentity, Errors>> completed = new HashMap<>();
+        final Map<CoordinatorKey, Throwable> failed = new HashMap<>();
+        final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();
+        final Set<CoordinatorKey> groupsToRetry = new HashSet<>();
 
-        final Errors error = Errors.forCode(response.data().errorCode());
+        final Errors error = response.topLevelError();
         if (error != Errors.NONE) {
-            handleError(groupId, error, failed, unmapped);
+            handleGroupError(groupId, error, failed, groupsToUnmap, groupsToRetry);
         } else {
             final Map<MemberIdentity, Errors> memberErrors = new HashMap<>();
             for (MemberResponse memberResponse : response.memberResponses()) {
+                Errors memberError = Errors.forCode(memberResponse.errorCode());
+                String memberId = memberResponse.memberId();
+
                 memberErrors.put(new MemberIdentity()
-                                     .setMemberId(memberResponse.memberId())
+                                     .setMemberId(memberId)
                                      .setGroupInstanceId(memberResponse.groupInstanceId()),
-                                 Errors.forCode(memberResponse.errorCode()));
+                    memberError);
```

Comment:
nit: We could revert this change as it does not bring much and re-align like it was before.